api/v1alpha2/conversion.go

```diff
@@ -247,9 +257,21 @@ func (dst *MachineDeployment) ConvertFrom(srcRaw conversion.Hub) error {
 		return err
 	}
 
+	// Manually convert annotations
+	for i := range v3Annotations {
+		convertAnnotations(v3Annotations[i], v2Annotations[i], src.Annotations)
```

Comment:
Shouldn't we convert `dst.Annotations` here? 

Also we should probably move this up before the MarshalData call

Reply:
Done